tests/countries/test_guinea.py

```diff
@@ -0,0 +1,162 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.guinea import Guinea, GN, GIN
+from tests.common import CommonCountryTests
+
+
+class TestGuinea(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(Guinea)
+
+    def test_country_aliases(self):
+        self.assertAliases(Guinea, GN, GIN)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(Guinea(years=1957))
+
+    def test_second_republic_day(self):
+        name = "Jour de la Deuxième République"
+        self.assertHolidayName(name, (f"{year}-04-03" for year in range(1958, 2021)))
+        self.assertNoHolidayName(name, Guinea(years=range(2022, 2050)))
+
+    def test_all_saints_day(self):
+        name = "La Toussaint"
+        self.assertHolidayName(name, (f"{year}-11-01" for year in range(1958, 2021)))
+        self.assertNoHolidayName(name, Guinea(years=range(2022, 2050)))
+
+    def test_new_years_day(self):
+        self.assertHolidayName("Nouvel an", (f"{year}-01-01" for year in range(1958, 2050)))
+
+    def test_africa_day(self):
+        self.assertHolidayName(
+            "Anniversaire de l'OUA", (f"{year}-05-25" for year in range(1958, 2050))
+        )
+
+    def test_eid_al_fitr_day(self):
+        name = "Korité"
+        self.assertHolidayName(
+            name,
+            "2015-07-18",
+            "2016-07-07",
+            "2017-06-26",
+            "2018-06-15",
+            "2019-06-04",
+            "2020-05-24",
+            "2021-05-13",
+            "2022-05-02",
+            "2023-04-21",
+            "2024-04-10",
+            "2025-03-31",
+        )
```

Comment:
For movable holidays, consider adding test to check if they're observed in all valid years as well:
```suggestion
    def test_eid_al_fitr_day(self):
        name = "Korité"
        self.assertHolidayName(
            name,
            "2015-07-18",
            "2016-07-07",
            "2017-06-26",
            "2018-06-15",
            "2019-06-04",
            "2020-05-24",
            "2021-05-13",
            "2022-05-02",
            "2023-04-21",
            "2024-04-10",
            "2025-03-31",
        )
        self.assertHolidayName(name, range(1959, 2050))
```